docs/runtime/reproducibility.md

```diff
@@ -124,3 +124,74 @@ You can omit the signature initially and add it later by using:
 ```bash
 orc sgx-set-sig bundle.orc path/to/binary.sig
 ```
+
+### Multi-step SGX Signing Example
+
+Multi-step signing allows enclave signing keys to be kept offline, preferrably
+in some HSM. The following example uses `openssl` and a locally generated key as
+an example, however, it is suggested that the key be stored in a more secure
+location than in plaintext on disk.
+
+#### Generate a key
+
+We will generate a valid key for enclave signing. This must be a
+3072-bit RSA key with a public exponent of 3. Do this like so:
+
+```bash
+openssl genrsa -3 3072 > private.pem
+```
+
+We will also need the public key in a later step so let's also generate this
+now.
+
+```bash
+openssl rsa -in private.pem -pubout > public.pem
+```
+
+#### Generate signing data for your enclave
+
+Generating signing data is done with the `orc sgx-gen-sign-data` subcommand,
+like so:
+
+```bash
+orc sgx-gen-sign-data [options] bundle.orc
+```
+
+_See `orc sgx-gen-sign-data --help` for details on available options_
+
+For purposes of this example, let's assume your bundle is named `bundle.orc`.
+You would generate data to sign like so:
+
+```bash
+orc sgx-gen-sign-data bundle.orc > sigstruct.sha256.bin
+```
+
+The output file `sigstruct.sha256.bin` contains the sha256 hash of the
+SIGSTRUCT fields to be signed.
+
+##### Sign the SIGSTRUCT hash
+
+To sign the SIGSTRUCT you must create a signature using the `RSASSA-PKCS1-v1_5`
+scheme. The following command will do so with `openssl`. If you're using an HSM,
+your device may have a different process for generating a signature of this
+type.
+
+```bash
+openssl pkeyutl -sign \
+      -in sigstruct.sha256.bin \
+      -inkey private.pem \
+      -out sigstruct.sha256.sig \
+      -pkeyopt digest:sha256
+```
+
+##### Attach the singed SIGSTRUCT to the bundle
+
+With the signature in `sigstruct.sha256.sig` we can now generate a valid
+SIGSTRUCT and attach it into the bundle.
+
+```bash
+orc sgx-set-sig bundle.orc sigstruct.sha256.sig public.pem
+```
+
+If there are no errors, `bundle.orc` will now contain a valid SGX SIGSTRUCT
```

Comment:
Does the SDK orc tool have a dump/validate sub-command that will open, verify, and dump the manifest in a nice human readable format?  If not, one should exist, and if such a thing exists, should show how to confirm that there is a SIGSTRUCT in the newly altered bundle.

Reply:
Yeah having an `orc show <bundle.orc>` would be really nice.

Reply:
Added the `show` subcommand and updated the docs.